src/OpenTelemetry.Api/Context/RuntimeContext.cs

```diff
@@ -52,10 +54,25 @@ public static void RegisterSlot<T>(string name)
 
                 var type = ContextSlotType.MakeGenericType(typeof(T));
                 var ctor = type.GetConstructor(new Type[] { typeof(string) });
-                Slots[name] = ctor.Invoke(new object[] { name });
+                var slot = (RuntimeContextSlot<T>)ctor.Invoke(new object[] { name });
+                Slots[name] = slot;
+                return slot;
             }
         }
 
+        /// <summary>
+        /// Get a registered slot from a given name.
+        /// </summary>
+        /// <param name="name">The name of the context slot.</param>
+        /// <typeparam name="T">The type of the underlying value.</typeparam>
+        /// <returns>The slot previously registered, or null if not found.</returns>
+        public static RuntimeContextSlot<T> GetSlot<T>(string name)
+        {
+            object slot;
+            Slots.TryGetValue(name, out slot);
+            return (RuntimeContextSlot<T>)slot;

```

Comment:
Will this cause a cast exception if the the TryGet fails?

Reply:
Yes it will. I expect consumers to call this once, and cache the slot for subsequent use.

Reply:
Perhaps I should just use `(RuntimeContextSlot<T>)Slots[name]`?

Reply:
I think that will throw a key not found. I'm just noting your comment
`<returns>The slot previously registered, or null if not found.</returns>`

Seems like the TryGetValue is a good way to go and just return null if that fails. Was that your intent?

Reply:
I've updated the code and also the PR description to show my current approach/thinking.

Reply:
👍 ok yea I get it, thanks.

Reply:
Also, the `using` idea is interesting - maybe we try on for size what you have here, and refactor if we like it?

Reply:
Yeah, I think we will have better understanding of how syntactical sugar will help exporters/processors/instrumentation when we start to add the code.